### PR TITLE
drivers: regulator: multiple cleanups/improvements

### PR DESCRIPTION
**Pull request by gmarull**
This is some prep-work for regulator API improvements. Summary of the changes:

- Moved "consumer.h" API to "regulator.h", and introduced driver ops.
- Renamed "regulator-pmic" to "nxp,pca9420", since it was a driver implementing support for NXP PCA9420 PMIC, not a "universal PMIC". Common fields/properties for PMICs will be refactored later, since this was the only PMIC driver in-tree.
- Doxygen improvements
- Other minor improvements.

If you were not aware, Nordic now makes PMICs as well: https://www.nordicsemi.com/Products/Power-Management. So, we'll need to move the regulator API out of its _toy_ state.

### CODE REVIEW

File: tests/drivers/regulator/pca9420/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 cmake_minimum_required(VERSION 3.20.0)
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
-project(tests_drivers_regulator_pmic)
+project(tests_drivers_regulator_pca9420)
```
Comment (danieldegrasse) [inline tests/drivers/regulator/pca9420/CMakeLists.txt:6]: Why move this test to be specific to the PCA9420? Is there something here that wouldn't work for another I2C compatible PMIC like the nPM6001?

**Review by danieldegrasse — COMMENTED** [review]
Thanks for taking this on. Just interested to know, are there significant shortcomings in the regulator API as it is implemented today (besides the issues this PR fixes). I tried to follow the Linux API whenever possible, but the API was only tested with a PCA9420.

File: tests/drivers/regulator/pca9420/CMakeLists.txt (hunk 1 above)
Comment (gmarull) [reply tests/drivers/regulator/pca9420/CMakeLists.txt:6]: I'm planning to improve tests as well, but right now, it is too specific IMO.

**Comment by gmarull** [comment]
> Thanks for taking this on. Just interested to know, are there significant shortcomings in the regulator API as it is implemented today (besides the issues this PR fixes). I tried to follow the Linux API whenever possible, but the API was only tested with a PCA9420.

It's not really shortcomings, API needs a bit of clarification (some interfaces are not clear or not consistent). We also miss stuff like events (e.g. overcurrent), automatic mode selection, etc. And the asynchronous part needs work as well. I hope that with additional drivers, we'll see the limitations/shortcomings more clearly (I haven't looked in detail to the NXP PMIC).

**Comment by danieldegrasse** [comment]
> It's not really shortcomings, API needs a bit of clarification (some interfaces are not clear or not consistent). We also miss stuff like events (e.g. overcurrent), automatic mode selection, etc. And the asynchronous part needs work as well. I hope that with additional drivers, we'll see the limitations/shortcomings more clearly (I haven't looked in detail to the NXP PMIC).

Makes sense. Interested to see what changes we need, and I'm glad to have another user of this API. One thing, could you update the [release notes](https://docs.zephyrproject.org/latest/releases/release-notes-3.3.html#new-apis-in-this-release) mentioning that these APIs are now present as part of the regulator API?

**Comment by gmarull** [comment]
> > It's not really shortcomings, API needs a bit of clarification (some interfaces are not clear or not consistent). We also miss stuff like events (e.g. overcurrent), automatic mode selection, etc. And the asynchronous part needs work as well. I hope that with additional drivers, we'll see the limitations/shortcomings more clearly (I haven't looked in detail to the NXP PMIC).
> 
> Makes sense. Interested to see what changes we need, and I'm glad to have another user of this API. One thing, could you update the [release notes](https://docs.zephyrproject.org/latest/releases/release-notes-3.3.html#new-apis-in-this-release) mentioning that these APIs are now present as part of the regulator API?

Done. Btw, if you guys could give a PCA9420 eval kit to do some testing, it'd be nice :-)

**Review by danieldegrasse — DISMISSED** [review]

File: drivers/regulator/regulator_pca9420.c [hunk 2]
```diff
@@ -36,8 +36,8 @@ struct regulator_config {
 	int num_modes;
 	uint8_t vsel_reg;
 	uint8_t vsel_mask;
-	uint32_t max_uV;
-	uint32_t min_uV;
+	uint32_t max_uv;
+	uint32_t min_uv;
```
Comment (carlocaione) [inline drivers/regulator/regulator_pca9420.c:40]: should this be `int32_t`?

Comment (gmarull) [reply drivers/regulator/regulator_pca9420.c:40]: good point, let me fix that

**Review by carlocaione — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**